### PR TITLE
feat: peer_connected hook chainable

### PR DESCRIPTION
**Pull request by m-schmoock**
Makes the `peer_connected` hook chainable.

Adds feature, doc and tests.

### CODE REVIEW

**Comment by cdecker** [comment]
I think the first "typo" commit was on purpose, done by @niftynei to differentiate the non-df from the df versions.

File: doc/PLUGINS.md [hunk 1]
```diff
@@ -777,7 +777,7 @@ no plugin is registered on the hook.
 ### `peer_connected`
 
 This hook is called whenever a peer has connected and successfully completed
-the cryptographic handshake. The parameters have the following structure if there is a channel with the peer:
+the cryptographic handshake. The parameters have the following structure:
```
Comment (cdecker) [inline doc/PLUGINS.md:780]: This seems unrelated.

**Comment by m-schmoock** [comment]
> I think the first "typo" commit was on purpose, done by @niftynei to differentiate the non-df from the df versions.

No, I just removed the "_" underscore character, not the "2" in openchannel2. The reason I did that was just to be more conistent with other hook log messages. "_hook" is not part of the name of the hook.

File: doc/PLUGINS.md (hunk 1 above)
Comment (m-schmoock) [reply doc/PLUGINS.md:780]: When reading the doc and reading the code that produces this DTO for the hook, the part "if there is a channel with the peer" seemed wrong to me, as the parameters will always be the same regardless if there is a channel or not. This is why I removed this part.

File: doc/PLUGINS.md [hunk 2]
```diff
@@ -789,7 +789,7 @@ the cryptographic handshake. The parameters have the following structure if ther
 }
 ```
 
-The hook is sparse on purpose, since the plugin can use the JSON-RPC
+The hook is sparse on information, since the plugin can use the JSON-RPC
```
[the remark this answers is not in the document]
Comment (m-schmoock) [reply doc/PLUGINS.md:792]: Well, I did that on purpose, as the 'purpose' of the hook is not sparse, just the 'information' that is sent to the hook ;)

**Comment by m-schmoock** [comment]
@cdecker  The tests to this PR run locally just fine. I will now add missing unit tests that test the chaining of this hook specifically. After this I will undraft this PR...

**Comment by m-schmoock** [comment]
Added tests and they are green also with `VALGRIND=1`. Undrafting the PR now.

File: doc/PLUGINS.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/PLUGINS.md:792]: In this case, "on purpose" means "deliberately", but the English grammar rule against split infinitives kind of applies here, so "The hook is on purpose sparse" sounds wrong.

Let's just use "deliberately"?

**Comment by rustyrussell** [comment]
Rebasing on master which now uses GH actions should now fix the CI breakage!

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/af8d56f2db0647f62e5a3cabdd45442702036cb4
